### PR TITLE
Locks/verify: full refspec

### PR DESCRIPTION
**Pull request by technoweenie**
Updates the `locks/verify` calls to send the full refspec. /cc #2712.

### CODE REVIEW

File: git/git.go [hunk 1]
```diff
@@ -95,6 +91,16 @@ type Ref struct {
 	Sha  string
 }
 
+func (r *Ref) Refspec() string {
```
Comment (ttaylorr) [inline git/git.go]: 👍 I agree with your comment that `Refspec` is a better name here. I think we can merge this as-is and I'll update my branch to pull in `Refspec()` over `String()`.

File: script/integration.go [hunk 2]
```diff
@@ -50,17 +50,31 @@ func mainIntegration() {
 	for _, file := range files {
 		tests <- file
 	}
+	close(tests)
+
+	outputDone := make(chan bool)
+	go func() {
```
Comment (ttaylorr) [inline script/integration.go]: Nit: this could be replaced with:

```go
go func() {
        for out := range output {
                fmt.Println(out)
        }
        outputDone <- true
}()
```

Though I am curious why this change is included in this PR?

Comment (technoweenie) [reply script/integration.go]: I was running into an issue where `script/integration` wouldn't dump out the `test-pre-push.sh` errors for this PR. This keeps the process alive until the tests have run AND the output has been sent STDOUT.

Comment (technoweenie) [reply script/integration.go]: I had planned on pushing that out as a separate PR. I must've accidentally merged to my local master, so it was still there when I branched off for this PR :)

I'll port the `Refspec()` change to #2734, and push out a PR for the test change, and another for this change.

Comment (ttaylorr) [reply script/integration.go]: Rad! Thanks for porting this into #2734, I think that should be ready to go now 👍 .

**Comment by technoweenie** [comment]
Don't merge. Will re-apply these changes once #2734 has merged to master.
